Source/WebCore/rendering/RenderBlock.cpp

```diff
@@ -2382,7 +2381,7 @@ void RenderBlock::computeBlockPreferredLogicalWidths(LayoutUnit& minLogicalWidth
         minLogicalWidth = std::max(w, minLogicalWidth);
 
         // IE ignores tables for calculation of nowrap. Makes some sense.
-        if (nowrap && !child->isTable())
+        if (!styleToUse.autoWrap() && !child->isTable())
```

Comment:
autowarp also consults PRE, is this behavior change intentional?

>text-wrap: nowrap is the correct thing to check for this reason.
Am I missing something here? on my local repo autowrap still uses whitespace.

Reply:
> autowarp also consults PRE, is this behavior change intentional?
`pre` also disables wrapping. From my understanding, anything which turns off wrapping can expand the max width?

I'd love your expert opinion here though.

> Am I missing something here? on my local repo autowrap still uses whitespace.
We'll want to change `RenderStyle::autoWrap()` to be `textWrap() != TextWrap::NoWrap`.

Reply:
> > autowarp also consults PRE, is this behavior change intentional?
> 
> `pre` also disables wrapping. From my understanding, anything which turns off wrapping can expand the max width?
> 
> I'd love your expert opinion here though.

I don't have a good understanding on the expected behavior here. The code was clearly checking against nowrap value only. Whether it is an existing bug or some quirky behavior, I have no clue. Digging through the history would certainly help to figure out (or coming up with a test case where Webkit fails vs. Blink/Gecko).

> 
> > Am I missing something here? on my local repo autowrap still uses whitespace.
> 
> We'll want to change `RenderStyle::autoWrap()` to be `textWrap() != TextWrap::NoWrap`.
Sure, but that's not what the current state of trunk is (and I believe the commit message is supposed to reflect that).

Reply:
> I don't have a good understanding on the expected behavior here. The code was clearly checking against nowrap value only. Whether it is an existing bug or some quirky behavior, I have no clue. Digging through the history would certainly help to figure out (or coming up with a test case where Webkit fails vs. Blink/Gecko).

Not much context, but it was introduced in this commit: https://commits.webkit.org/3134@main

> Sure, but that's not what the current state of trunk is (and I believe the commit message is supposed to reflect that).

I'll update the commit message.